doc-site/docs/tutorials/public-storage.md

```diff
@@ -0,0 +1,131 @@
+
+```typescript
+const deploymentTxID = await paladin.sendTransaction({
+  type: TransactionType.PUBLIC,      // Public deployment
+  abi: storageJson.abi,              // ABI of the Storage contract
+  bytecode: storageJson.bytecode,    // Compiled bytecode
+  function: "",                      // No constructor arguments
+  from: owner.lookup,                // Account signing the transaction
+  data: {},                          // No additional data
+});
+
+const deploymentReceipt = await paladin.pollForReceipt(deploymentTxID, 10000);
+if (!deploymentReceipt?.contractAddress) {
+  logger.error("Deployment failed!");
+  return false;
+}
+logger.log("Step 1: Storage contract deployed successfully!");
+```
+
+- **What happens**:
+  - The `sendTransaction` function creates a deployment transaction for the `Storage` contract.
+  - The `pollForReceipt` function waits for the transaction to be confirmed.
+  - On success, the contract address is returned in the receipt.
+
+---
+
+### Step 2: Store a Value
+
+After deploying the contract, you can store a value in the contract using its `store` function.
+
+```typescript
+const valueToStore = 125; // Example value to store
+logger.log(`Step 2: Storing value "${valueToStore}" in the contract...`);
+
+const storeTxID = await paladin.sendTransaction({
+  type: TransactionType.PUBLIC,       // Public transaction
+  abi: storageJson.abi,               // ABI of the Storage contract
+  function: "store",                  // Name of the function to call
+  from: owner.lookup,                 // Account signing the transaction
+  to: deploymentReceipt.contractAddress, // Address of the deployed contract
+  data: { num: valueToStore },        // Function arguments
+});
+
+const storeReceipt = await paladin.pollForReceipt(storeTxID, 10000);
+if (!storeReceipt?.transactionHash) {
+  logger.error("Failed to store value in the contract!");
+  return false;
+}
+logger.log("Step 2: Value stored successfully!");
+```
+
+- **What happens**:
+  - The `sendTransaction` function sends a transaction to call the `store` function with the value to store (`125` in this example).
+  - The `pollForReceipt` function waits for the transaction to be confirmed.
+
+---
+
+### Step 3: Retrieve the Stored Value
+
+Next, retrieve the stored value using the `retrieve` function of the contract.
+
+```typescript
+logger.log("Step 3: Retrieving the stored value...");
+
+const retrieveResult = await paladin.call({
+  type: TransactionType.PUBLIC,       // Public call
+  abi: storageJson.abi,               // ABI of the Storage contract
+  function: "retrieve",               // Name of the function to call
+  from: owner.lookup,                 // Account making the call
+  to: deploymentReceipt.contractAddress, // Address of the deployed contract
+  data: {},                           // No arguments required for this function
+});
+
+const retrievedValue = retrieveResult["0"];
```

Comment:
I commented on the smart contract, but we may want to encourage naming the return value something other than `0`?